kiali/kiali
# [Tempo] Add Tempo cache

### Describe the change

Adds a TTL cache for Tempo to minimize the access to the Tempo backend. 

It creates a couple of structures: 
- One that saves the results for Tempo queries
- Other that saves the result for Tempo individual traces

The cache for queries save the results by minutes granularity, trying to increase the cache effectiveness. 

As a TTL cache, every entry has a expiration time that removes the entry once is expired. 

This expiration time is defined globally for all the elements (CacheExpiration) and it is set to 300s by default. It will be checked by a watcher every minute, as the watcher needs to check all the entries so decrease the watcher time, will probably have a higher cost. 

Every time the watcher runs, it will display the hit rate (Since the cache was created) and the cache size. So it might help to adjust the expiration time. 

This cache can be useful, for example, when the user is looking at the traces, and click on the same trace twice (It can be done for comparison). Also, when it clicks on the traffic graph node, in the traces tab, and click back on the same node. Here it can save 10 calls to the backend.

### Steps to test the PR

Run the environment with: 
- `hack/istio/tempo/install-tempo-env.sh -c kubectl`
- Install Kiali
- Tempo cache should be enabled 

### Automation testing
Covered in Tempo tracing CI pipeline

### Issue reference

Ref. #7769
